services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/Device.java

```diff
@@ -38,6 +40,12 @@
     @JsonProperty(RegistryManagementConstants.FIELD_ENABLED)
     private Boolean enabled;
 
+    @JsonProperty(RegistryManagementConstants.FIELD_AUTO_PROVISIONED)
+    private Boolean autoProvisioned;
+
+    @JsonProperty(RegistryManagementConstants.FIELD_AUTO_PROVISIONING_NOTIFICATION_SENT)
+    private Boolean autoProvisioningNotificationSent;
```

Comment:
IMHO these should better be moved to the device's *status* property because we do not want clients to be able to update these, right?

Reply:
Yes that it is right. I looked at the implementation of the status property and it seems like it is not complete yet. Consequently on my local deployment (with file based registry) there is only the current time returned as created and updated timestamp.

There is also an open issue regarding this: https://github.com/eclipse/hono/issues/1960

So I guess it makes sense to implement the status property first and then come back to this PR?

Reply:
Either that or do not include the fields in the device object being transferred between the client and the registry

Reply:
I am struggling with what you mean with

| do not include the fields in the device object being transferred between the client and the registry

I was looking for some kind of "internal model" initially already but went to re-investigate if there is such a thing after you remark. As far as I can see the `org.eclipse.hono.service.management.device.Device` class is used as the sole model class - internally as well as for the API. So the only option I see to have those two fields persistent is to include them in that class.

Did I miss somehing?

Reply:
No, you didn't. However, that doesn't mean that you need to transfer their values hence and forth between the registry and the client, does it?

Reply:
Okay, so I guess you mean `FileBasedDeviceBackend` and `MongoDbBasedDeviceBackend` with _client_.

The problem is that we do a replace of the device when persisting for the file-based registry (there we just replace the entry in the HashMap) as well as for the MongoDb-based registry (there we replace the document of the device).

If I do not transfer the values (i.e. setting them to _null_) the present data will be overwritten. In order to put this to work I would need to implement some kind of "merge" as far as I can see.

But I do think that the status property is a better solution than implementing merge-logic (which is just a workaround for the missing status property in my opinion).

Hence I do actually only see the implementation of the status property as a solution. So I would continue there and come back to this PR.

WDYT?

Reply:
Finally https://github.com/eclipse/hono/pull/2189 is near being merged. After it was merged I am able to continue on this one.

I guess it makes sense to squash my commits to the present one and then continue with moving the auto-provisioning properties to the device status in a separate commit. WDYT?